src/global.rs

```diff
@@ -1,46 +1,51 @@
+use crate::AsyncRayonHandle;
+use std::panic::{catch_unwind, AssertUnwindSafe};
 use tokio::sync::oneshot;
-use tokio::sync::oneshot::error::RecvError;
 
 /// Asynchronous wrapper around Rayon's [`spawn`](rayon::spawn).
 ///
 /// Runs a function on the global Rayon thread pool with LIFO priority,
 /// produciing a future that resolves with the function's return value.
 ///
-/// # Errors
-/// Forwards Tokio's [`RecvError`](tokio::sync::oneshot::error::RecvError), i.e. if the channel is closed.
-pub async fn spawn_async<F, R>(func: F) -> Result<R, RecvError>
+/// # Panics
+/// If the task function panics, the panic will be propagated through the
+/// returned future. Thie will NOT trigger the Rayon thread pool's panic
+/// handler.
+pub fn spawn_async<F, R>(func: F) -> AsyncRayonHandle<R>
```

Comment:
I'm not sure the `spawn_async` name is great. You aren't spawning an async task here.  Maybe just call it `spawn`?

Reply:
`spawn` on the `AsyncThreadPool` trait conflicts with `spawn` on `rayon::ThreadPool`, and I wanted to use the same name for the function like Rayon does. It's an async wrapper around `spawn` I guess?

Reply:
The more I think about it, I do like `spawn` better, and the trait methods can just be named differently :)